src/components/model-authoring/glossary-settings.tsx

```diff
@@ -12,7 +15,17 @@ interface IProps {
 }
 
 export const GlossarySettings = ({ name, glossary, saveSettings, saveName }: IProps) => {
-  const { askForUserDefinition, showSideBar, autoShowMediaInPopup, enableStudentRecording, enableStudentLanguageSwitching } = glossary;
+  const { askForUserDefinition, showSideBar, autoShowMediaInPopup,  enableStudentRecording, enableStudentLanguageSwitching } = glossary;
+  const [disabledOptions, setDisabledOptions] = useState<boolean>(askForUserDefinition);
+
+  const handleUserDefinitionChange = () => (e: React.ChangeEvent<HTMLInputElement>) => {
+    setDisabledOptions(e.target.checked);
+    if (!e.target.checked){
+      saveSettings({...glossary, "askForUserDefinition": e.target.checked, "enableStudentRecording": false})
```

Comment:
I think this should be `"askForUserDefinition": !e.target.checked` or just `"askForUserDefinition": false`